peer.go

```diff
@@ -1775,6 +1762,23 @@ out:
 			// relevant sub-systems and launching a goroutine to
 			// wait for close tx conf.
 			p.finalizeChanClosure(chanCloser)
+
+		// The channel reannounce delay has elapsed, broadcast the
+		// reenabled channel updates to the network. This should only
+		// fire once, so we set the reenableTimeout channel to nil to
+		// mark it for garbage collection. If the peer is torn down
+		// before firing, reenabling will not be attempted.
+		case <-reenableTimeout:
```

Comment:
I'm wondering if it would be even nicer to move this logic to the `ChanStatusManager`. That way the peer/server would only notify the `ChanStatusManager` about changes to the active status of the link: `becameActive`/`becameInactive`, and then the `manager` would make sure to wait for a full interval duration without any inactive events for the link before re-enabling it.

Reply:
To keep the current behaviour, it would send `becameActive` when entering the `channelManager`, and `becameInactive` when exiting.

Reply:
I thought about this, and ultimately decided it wasn't worth it at this point in time. if we do so, we then need extra goroutines/timers inside the chan status manager to properly time the reenable, instead of reusing the existing peer's resources/lifecycle to govern this. it would be nice to have this fully self-contained, tho I think it will add a fair amount of additional complexity to the state machine and bookkeeping. Given how much time has already been spent on this refactor, I would vote to defer this

Reply:
I agree. Should add a todo for this.